youtube_dl/extractor/bbc.py

```diff
@@ -959,6 +959,32 @@ def _real_extract(self, url):
              r'videoId\s*:\s*["\'](%s)["\']' % self._ID_REGEX],
             webpage, 'vpid', default=None)
 
+        # bbc reel (e.g. https://www.bbc.com/reel/video/p07c6sb6/how-positive-thinking-is-harming-your-happiness)
+        if not programme_id:
+            initial_data = self._search_regex(
+                r'<script[^>]+id="initial-data"[^>]+data-json=\'(.+)\'>',
```

Comment:
this can match outside the script element and can be easily broken if an attribute follow `data-json`.

Reply:
So ajj8 should have known about `_html_search_regex` and the regex attribute matching should at least use `/attr='[^']+'/`. I've given it what should be exact according to W3C HTML5 8.1.2, which might be wise as they're already using both "" and '' attribute values in the same tag.

https://github.com/dirkf/youtube-dl-1/commit/2dcd5c9437d20d43b94180ae57d4d0807f583cdc